### PR TITLE
[Platformer] Fix: platformer characters can now jump when moving against a wall that is part of the same object as the floor.

### PR DESCRIPTION
**Pull request by D8H**
This could happen when using a tilemap collision mask object and not sustaining the jump.

It's covered by tests but can also be reproduced on this example:
* https://github.com/GDevelopApp/GDevelop-examples/pull/343

### CODE REVIEW

**Comment by D8H** [comment]
This fix has a side effect: with a very low jump speed while climbing a slope, the character can no longer jump.

File: Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts [hunk 1]
```diff
@@ -1723,14 +1723,15 @@ namespace gdjs {
       behavior._requestedDeltaX +=
         this._floorPlatform!.owner.getX() - this._floorLastX;
       // See `beforeUpdatingObstacles` for the logic for the Y axis.
+
+      //Jumping
+      behavior._checkTransitionJumping();
```
Comment (4ian) [inline Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts:1728]: Can you explain in plain English why this change enabled to fix the issue? :)

**Review by 4ian — APPROVED** [review]

File: Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts (hunk 1 above)
Comment (ClementPasteau) [reply Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts:1728]: +1 I'm not sure I understand why this needs to happen in the beforeMovingX, and not in the checkTransitionBeforeY anymore?
Also, should it happen in the beforeMovingY() too?

Comment (D8H) [reply Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts]: Sorry I meant to move it in `checkTransitionBeforeX` not `beforeMovingX`. I added a comment.

Comment (D8H) [reply Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts]: > Also, should it happen in the beforeMovingY() too?

The check is about the jump key. It won't change during `doStepPreEvents`.

File: Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts [hunk 2]
```diff
@@ -1715,6 +1715,17 @@ namespace gdjs {
...
+
+      // It was originally in checkTransitionBeforeY.
+      // The character is ignoring the floor when moving on X to be able to
+      // follow up a slope when moving Y (it enter inside it).
+      // When the current floor and the wall the character is facing is part of
+      // the same instance, the wall is also ignored when moving on X, but the
+      // wall is to high to follow and it is seen as colliding an obstacle
+      // from behind.
+      // Moving against a wall before jumping in this configuration was making
+      // jumps being aborted.
```
Comment (D8H) [inline Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts:1727]: I added this comment.

Comment (ClementPasteau) [reply Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts:1727]: nice! just one typo: "to high" => "too high"

**Review by ClementPasteau — APPROVED** [review]
